src/ipa_pc/mod.rs

```diff
@@ -802,22 +1095,54 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
             opening_challenge_counter += 1;
         }
 
+        let key_len = ck.comm_key.len();
+        let p_len = batch_polynomial.coeffs.len();
+        let segments_count = std::cmp::max(1, p_len / key_len + if p_len % key_len != 0 { 1 } else { 0 });
```

Comment:
Why the max here? To cover the case that `p_len = 0`?

Reply:
Yes, exactly. I've got this case in one of UTs